src/test/java/org/opensearch/ad/bwc/ADBackwardsCompatibilityIT.java

```diff
@@ -83,25 +131,176 @@ public void testBackwardsCompatibility() throws Exception {
+                    assertNotNull(realtimeTask);
+                    assertFalse(realtimeTask.isDone());
+                }
+            }
+            if (realtimeTaskMissing) {
+                Thread.sleep(10_000);// sleep 10 seconds to wait for realtime job to backfill realtime task
+            }
+        } while (realtimeTaskMissing && i < maxRetryTimes);
+    }
+
+    private void stopAndDeleteDetectors() throws Exception {
+        for (String detectorId : runningRealtimeDetectors) {
+            deleteRunningDetector(detectorId);
+            Response stopRealtimeJobResponse = stopRealtimeJob(client(), detectorId);
+            assertEquals(RestStatus.OK, TestHelpers.restStatus(stopRealtimeJobResponse));
+            try {
+                Response stopHistoricalAnalysisResponse = stopHistoricalAnalysis(client(), detectorId);
+                assertEquals(RestStatus.OK, TestHelpers.restStatus(stopHistoricalAnalysisResponse));
+            } catch (Exception e) {
+                if (!ExceptionUtil.getErrorMessage(e).contains("No running task found")) {
+                    throw e;
+                }
+            }
+            Map<String, Object> jobAndTask = getDetectorWithJobAndTask(client(), detectorId);
+            AnomalyDetectorJob job = (AnomalyDetectorJob) jobAndTask.get(ANOMALY_DETECTOR_JOB);
+            ADTask historicalAdTask = (ADTask) jobAndTask.get(HISTORICAL_ANALYSIS_TASK);
+            if (!job.isEnabled() && historicalAdTask.isDone()) {
+                Response deleteDetectorResponse = deleteDetector(client(), detectorId);
+                assertEquals(RestStatus.OK, TestHelpers.restStatus(deleteDetectorResponse));
+            }
+        }
+    }
+
+    private void startHistoricalAnalysisOnNewNode(String detectorId, String taskType) throws IOException, InterruptedException {
+        String taskId = startHistoricalAnalysis(client(), detectorId);
+        deleteRunningDetector(detectorId);
+        waitUntilTaskDone(client(), detectorId);
+        List<ADTask> adTasks = searchLatestAdTaskOfDetector(client(), detectorId, taskType);
+        assertEquals(1, adTasks.size());
+        assertEquals(taskId, adTasks.get(0).getTaskId());
+        int adResultCount = countADResultOfDetector(client(), detectorId, taskId);
+        assertTrue(adResultCount > 0);
+    }
+
+    private void startRealtimeJobForHistoricalDetectorOnNewNode() throws IOException {
+        for (String detectorId : historicalDetectors) {
+            String jobId = startAnomalyDetectorDirectly(client(), detectorId);
+            assertEquals(detectorId, jobId);
+            Map<String, Object> jobAndTask = getDetectorWithJobAndTask(client(), detectorId);
+            AnomalyDetectorJob detectorJob = (AnomalyDetectorJob) jobAndTask.get(ANOMALY_DETECTOR_JOB);
+            assertTrue(detectorJob.isEnabled());
+            runningRealtimeDetectors.add(detectorId);
+        }
+    }
+
+    private void verifyRealtimeJobRunning() throws IOException {
```

Comment:
nit: maybe rename to indicate that this is verifying **all current** realtime jobs?

Reply:
sure